### ISSUE TITLE
.text on element does not return text content of child button element

### ISSUE THREAD

**Issue by ldabiralai** [pzavolinsky/react-unit#12]
I'm using ```findByQuery``` to target a span, which contains a child button element. Then I try to assert on the text of the span, but it comes back as an empty string with the text of the button omitted.

Is this expected behaviour? It seems a bit strange to me.

I've written a snippet of code which replicates what I'm talking about:

```javascript
import React from 'react'
import createComponent from 'react-unit'
import {expect} from 'chai'

class TestComponent extends React.Component {
  render() {
    return (
      <span>
        <button>Hello World</button>
      </span>
    )
  }
}

describe('Component', () => {
  it('should be able to assert text in child element', () => {
    const component = createComponent.shallow(<TestComponent />);
    const buttonWrapper = component.findByQuery('span')[0];
    const button = component.findByQuery('button')[0];

    expect(button.text).to.eq('Hello World');
    expect(buttonWrapper.text).to.eq('Hello World');
  });
});
```

Here's the output when run, line 22 being the second expect:

![image](https://cloud.githubusercontent.com/assets/1711610/11574598/7df010ba-9a04-11e5-99cb-1bc39b667b49.png)

**Comment by bdo** [comment]
Hi @ldabiralai, I had a look and it seems that a few HTML elements (including button) are currently excluded for text generation:

```javascript
// Text functions
var excludeTextFrom = ['option', 'optgroup', 'textarea', 'button'];
```

Removing the code above fixes your unit test.

@pzavolinsky Should we not change this as this behaviour in different from what the DOM would do?

**Comment by pzavolinsky** [comment]
Hi @ldabiralai and @bdo you are both right, this is a (carefully crafted) bug. Originally, `.text` was meant to be a device for asserting the text of *some* elements, when it really should've been similar to `DOMNode.textContent` all along.

@ldabiralai thanks for pointing this out
@bdo thanks for the solution :)

Just pushed the fix in `1.1.1`

Cheers!

**Comment by ldabiralai** [comment]
Thanks for the quick response and fix @pzavolinsky!
